model_bakery/utils.py

```diff
@@ -21,6 +23,22 @@ def import_from_str(import_string: Optional[Union[Callable, str]]) -> Any:
         return import_string
 
 
+def get_calling_module(levels_back: int) -> Optional[ModuleType]:
+    """Get the module some number of stack frames back from the current one.
+
+    Make sure to account for the number of frames between the "calling" code
+    and the one that calls this function.
+
+    Args:
+        levels_back (int): Number of stack frames back from the current
+
+    Returns:
+        (ModuleType): the module from which the code was called
+    """
+    frame = inspect.stack()[levels_back + 1][0]
+    return inspect.getmodule(frame)
```

Comment:
There's a function in `utils.py` named [`import_from_str`](https://github.com/model-bakers/model_bakery/blob/main/model_bakery/utils.py#L10). It's used by `baker.make` or `baker.prepare` to work with string paths instead of model classes. 

This PR looks great and it definitely makes bakery more stable in terms of its external API. But I wonder if we could re-use the previous import function instead of defining new strategies to import models based on strings.

Reply:
So, this functionality was already present in the [import logic of `RecipeForeignKey`](https://github.com/model-bakers/model_bakery/blob/main/model_bakery/recipe.py#L76) and [of `related`](https://github.com/model-bakers/model_bakery/blob/main/model_bakery/recipe.py#L103). I simply extracted it out into a re-usable function.

I personally think the logic here (using stack frames to get the module from which the code was called) is a bit fragile, but it was already the way it was implemented so I left it alone (aside from pulling it into its own function to avoid duplication).

I am definitely open to alternatives, if you think there is a better way to access the "calling module" and its attributes. Just let me know what you think, and I will update the code. Thanks!

Reply:
For context, I did re-use the `baker._recipe` import function (which calls `import_from_str`) for [the new logic that I added](https://github.com/model-bakers/model_bakery/blob/175e3ea8c4b46a2d1ad13d8be9b2e2080ea9680f/model_bakery/recipe.py#L113), however, I left the "import from calling module" logic as a fallback (to maintain existing use of the `foreign_key` function).

Reply:
@berinhard Please let me know if there is a way to facilitate loading a recipe from the module that called `foreign_key` that can re-use any existing function.

Reply:
@timjklein36 sorry I didn't have the time to properly review this last week. But I'll do so today and will update the PR with comments.